source/features/quick-repo-deletion.tsx

```diff
@@ -22,12 +34,122 @@ async function isRepoUnpopular(): Promise<boolean> {
+		type: 'error',
+		action: (
+			<a className="btn btn-sm primary flash-action" href="https://github.com/settings/tokens">
+				Update token…
+			</a>
+		),
+	});
+}
+
+function setButtonText(button: DeleteButton, text: string): void {
+	if (button instanceof HTMLAnchorElement) {
+		button.textContent = text;
+	} else {
+		const label = $('.Button-label', button)!;
+		label.textContent = text;
+	}
+}
+
+function removeButtonIfAtRepoRoot(button: DeleteButton): void {
+	if (button instanceof HTMLAnchorElement) {
+		const buttonContainer = button.closest('li')!;
+		buttonContainer.remove();
+	}
+}
+
+async function notifyDeletionFailure(error: Error): Promise<void> {
+	await addNotice([
+		'Could not delete the repository. ',
+		(error as RefinedGitHubAPIError).response?.message ?? error.message,
+	], {
+		type: 'error',
+	});
+}
+
+async function deleteRepository(nameWithOwner: string): Promise<void> {
+	await api.v3('/repos/' + nameWithOwner, {
+		method: 'DELETE',
+		json: false,
+	});
+}
+
+async function modifyUIAfterSuccessfulDeletion(): Promise<void> {
+	const {nameWithOwner, owner} = getRepo()!;
+	const forkSource = '/' + getForkedRepo()!;
+	const restoreURL = pageDetect.isOrganizationRepo()
+		? `/organizations/${owner}/settings/deleted_repositories`
+		: '/settings/deleted_repositories';
+	const otherForksURL = `/${owner}?tab=repositories&type=fork`;
+
+	await addNotice(
+		<>
+			<TrashIcon />
+			<span>
+				Repository <strong>{nameWithOwner}</strong> deleted. <a href={restoreURL}>Restore it</a>, <a href={forkSource}>visit the source repo</a>, or see <a href={otherForksURL}>your other forks.</a>
+			</span>
+		</>,
+		{action: false},
+	);
+	$('.application-main')!.remove();
+
+	if (document.hidden) {
+		// Try closing the tab if in the background. Could fail, so we still update the UI above
+		void messageBackground({closeTab: true});
+	}
+}
+
+async function performDeletion(button: DeleteButton): Promise<void> {
+	if (!(await tokenHasDeleteRepoScope())) {
+		notifyMissingTokenScope();
+		return;
+	}
+
+	const originalButtonText = button.textContent;
+	const {nameWithOwner} = getRepo()!;
+
+	setButtonText(button, 'Deleting repo…');
+
+	try {
+		deleteRepository(nameWithOwner);
+	} catch (error) {
+		assertError(error);
+		removeButtonIfAtRepoRoot(button);
+		notifyDeletionFailure(error);
+
+		throw error;
+	}
+
+	setButtonText(button, originalButtonText);
+	modifyUIAfterSuccessfulDeletion();
+}
+
+function deleteButtonClicked(event: RepoRootClickEvent | RepoSettingsClickEvent): void {
+	if (event.ctrlKey && event.altKey && confirm('Are you sure you want to delete this fork?')) {
+		event.preventDefault();
+		performDeletion(event.target as HTMLAnchorElement | HTMLButtonElement);
+	}
+}
+
 function addButton(header: HTMLElement): void {
 	header.prepend(
 		<li>
 			<a
 				href={buildRepoURL('settings', buttonHashSelector)}
 				className="btn btn-sm btn-danger"
+				onClick={deleteButtonClicked}
```

Comment:
You should use delegate too, that's what we usually do. GitHub generates and regenerates the DOM losing listeners.